rs/p2p/state_sync_manager/src/ongoing.rs

```diff
@@ -259,26 +272,42 @@ impl OngoingStateSync {
                         &self.rt,
                     );
                 }
-                Some(_) => {}
-                None => {
-                    // If we store chunks in self.chunks_to_download we will eventually initiate  and
-                    // by filtering with the current in flight request we avoid double download.
-                    // TODO: Evaluate performance impact of this since on mainnet it is possible
-                    // that `chunks_to_download` returns 1Million elements.
-                    let mut v = Vec::new();
-                    for c in tracker.lock().unwrap().chunks_to_download() {
-                        if !self.downloading_chunks.contains(&c) {
-                            v.push(c);
-                        }
-                    }
-                    self.metrics.chunks_to_download_calls_total.inc();
-                    self.metrics.chunks_to_download_total.inc_by(v.len() as u64);
-                    self.chunks_to_download = Box::new(v.into_iter());
-                }
+                None => break,
             }
         }
     }
 
+    fn update_chunks_to_download<T: 'static + Send>(
+        &mut self,
+        tracker: Arc<Mutex<Box<dyn Chunkable<T> + Send>>>,
```

Comment:
nit: the `Arc` is unnecessary IIUC
```suggestion
        tracker: &Mutex<Box<dyn Chunkable<T> + Send>>,
```